src/sage/schemes/elliptic_curves/ell_rational_field.py

```diff
@@ -2333,7 +2426,26 @@ def _compute_gens(self, proof,
             except RuntimeError:
                 pass
         # end if (not_use_mwrank)
-        if algorithm == "mwrank_lib":
+        if algorithm == "pari":
```

Comment:
This block repeats lines 2210-2227 rather closely.  Would it be possible to have a helper function used by both?  The helper would not itself raise a runtime error, but that condition would be caught by the calling function.

Reply:
Do you insist? I am too lazy to put it into one for just two cases. I agree it might be better.